src/js/containers/addData/AddDataContainer.jsx

```diff
@@ -41,27 +42,62 @@ class AddDataContainer extends React.Component {
 		if (kGlobalConstants.LOCAL == true) {
 			UploadHelper.performLocalUpload(this.props.submission)
 				.then((submissionID) => {
-					// TODO: Remove this when this is eventually tied to user accounts
 					this.props.setSubmissionId(submissionID);
-					hashHistory.push('/validateData/' + submissionID);
-
+					// Looping because we need to allow backend to catchup to front end and prevent incorrect 404
+					let count = 10;
+					for(let i = 0; i < count; i++){
+						GuideHelper.getSubmissionPage(submissionID)
+							.then((res) => {
+								hashHistory.push('/validateData/' + submissionID);
+							})
+							.catch((err) => {
+								if(i == count - 1) {
```

Comment:
Not that it's a huge deal, but if we want to limit it to 10 we can either start i at 1 or set count to 9 rather than doing this calculation every time. It's not an intensive calculation, but why do any extra processing when we don't need to?